### ISSUE TITLE
Drop frame in exoplayer if player buffer size becomes greater than 200ms

### ISSUE THREAD

**Issue by Akansha-Yadav** [google/ExoPlayer#5356]
I am playing tcp stream on exoplayer to show the desktop screen on player.
My requirement is to drop frame or reduce player buffer if the buffer becomes greater than some given value say 300ms to reduce the latency in showing the desktop screen on player.

I searched in the code ( MediaCodecRenderer )for input buffers but this class is storing the input buffer for decoder. I need the exact player buffer which i can reduce to drop one complete frame in exoplayer if buffers size becomes greater than some given value or some other way to drop the frame on the basis of player buffer size.

Please help  me with this.

**Comment by andrewlewis** [comment]
You can reduce the amount of media that needs to be buffered for playback to start when building `DefaultLoadControl`. If the player falls too far behind the live edge can you just seek forwards?

**Comment by Akansha-Yadav** [comment]
Thanks for the reply.
I tried with the following configuration for DefaultLoadControl:
minBufferMs= 1000
maxBufferMs= 1000
bufferForPlaybackMs= 200

But still when i check for player.getBufferedPosition() - player.getCurrentPosition() after every one second, this value keep  on increasing and reaches to 1000,2000.....so on while running on default playback speed i.e. 1f

when i make the playback speed following condition based:

if (player.getBufferedPosition() - player.getCurrentPosition() > 100) {
                player.setPlaybackParameters(new PlaybackParameters(1.25f, 1));
            } else {
                player.setPlaybackParameters(new PlaybackParameters(1f, 1));
            }

It will increase, decrease and keeps repeating this cycle.
My understanding on this, whenever the buffer becomes greater than 100ms it will play the buffer at 1.25f ( fast speed than default ) and the buffer  decrease and playback speed will become 1f, this will again increase the buffer, and this cycle repeats.

What i need is exoplayer will never keep the buffer more than given duration say 200ms.  If its buffer becomes greater than 200ms it will  drop the frame while  playing at default 1f speed.

I also tried to seek the stream forward.  But no luck on this too. I was not able to seek the stream forward.

Stream Info: 
Stream protocol= TCP
Format= mpegts
Extension= ts
